### ISSUE TITLE
Supports scheduling jobs using `Attribute` instead of configuration files or programming.

### ISSUE THREAD

**Issue by fudiwei** [quartznet/quartznet#1268]
As the title says, I hope that Job can be defined as follows:

```csharp
[QuartzJob("* * * * * ? ", Description = "Tell the time every seconds.")]
public class ClockingJob : IJob
{
    private readonly ILogger _logger;

    public ClockingJob(ILoggerFactory loggerFactory)
    {
        _logger = loggerFactory.CreateLogger(GetType());
    }

    Task IJob.Execute(IJobExecutionContext context)
    {
        _logger.LogInformation("It's {0} now.", context.FireTimeUtc);
        return Task.CompletedTask;
    }
}
```

So I developed a library [Quartz.Extensions.JobAttribute](https://github.com/fudiwei/Quartz.Extensions.JobAttribute).

Can this be a feature?

**Comment by lahma** [comment]
Sounds like a cool feature. Some things that spring into my mind:

* should have separate attribute per trigger type, e.g. `[CronSchedule("* 1 * *")]`
* require sort of plugin/processor to scan for types, basically XmlSchedulingPlugin using attributes

**Comment by fudiwei** [comment]
> require sort of plugin/processor to scan for types, basically `XmlSchedulingPlugin` using attributes.

Did you mean `XMLSchedulingDataProcessorPlugin`?

**Comment by lahma** [comment]
Yes, `XMLSchedulingDataProcessorPlugin` could probably work as base like it's being used in MS DI integration to schedule gathered items. The implementation would be responsible for scanning the attributes. I think this attribute based implementation could reside in Quartz.Plugins project.

**Comment by lahma** [comment]
This feature hasn't received much comments and I think it can be a bit confusing to tie the job so tightly to its schedule.  There's custom library so that can be used for now. Thank you bringing the idea on the table though.
